src/lib/homestore_backend/hs_blob_manager.cpp

```diff
@@ -363,14 +363,22 @@ void HSHomeObject::on_blob_del_commit(int64_t lsn, sisl::blob const& header, sis
 
     auto r = move_to_tombstone(index_table, blob_info);
     if (!r) {
-        LOGW("fail to move blob to tombstone,  blob_id {}, shard_id {}, {}", blob_info.blob_id, blob_info.shard_id,
-             r.error());
-        if (ctx) ctx->promise_.setValue(folly::makeUnexpected(r.error()));
-        return;
+        if (recovery_done_) {
```

Comment:
Why is it an error if we are not in recovery, but success if we are? Shouldn't it just be success regardless? Maybe I don't understand exactly when `move_to_tombstone` could fail...

Reply:
We return error if delete failed to mark it as tombstone in usual rpc request. Only for recovery we can ignore the error. move_to_tombstone if there is any issue in accessing index or delete came again for a GCed blob etc. rare though.

Reply:
I guess I just want to make sure that deleting a BLOB in *any* case where a read would not be able to read the blob should return success. This includes when the blob never existed, or previously existed and no does not. As long as both those would be *success* on blob delete I'm okay with this...we should only fail when a subsequent read would return a blob.

Reply:
If i understood you correctly, yeah a duplicate delete should be success